### PR TITLE
Consolidate .netcoreapp.cs files in Common project

### PR DESCRIPTION
**Pull request by Marusyk**
Contributes dotnet/corefx#41821

Please review
Thank you in advance

### CODE REVIEW

**Comment by jkotas** [comment]
> src\libraries\Common\tests\System\Collections\IDictionary.NonGeneric.Tests.cs(841,32): error CS1061: 'DictionaryEntry' does not contain a definition for 'Deconstruct' and no accessible extension method 'Deconstruct' accepting a first argument of type 'DictionaryEntry' could be found (are you missing a using directive or an assembly reference?) 

The Common project is building for both .NET Core and .NET Framework desktop. Some of this split will need to stay to avoid these build breaks.

File: src/libraries/System.ComponentModel.TypeConverter/tests/TypeDescriptorTests.cs [hunk 1]
```diff
@@ -1140,6 +1140,12 @@ public void DerivedPropertyAttribute()
             Assert.Equal("Derived", descriptionAttribute.Description);
         }
 
+        [Theory]
+        [InlineData(typeof(Version), typeof(VersionConverter))]
+        [InlineData(typeof(IComponent), typeof(ComponentConverter))]
```
Comment (jkotas) [inline src/libraries/System.ComponentModel.TypeConverter/tests/TypeDescriptorTests.cs]: Fold these into the list on the existing `GetConverter` method.

File: src/libraries/System.ComponentModel.TypeConverter/tests/TypeDescriptorTests.cs [hunk 2]
```diff
@@ -1140,6 +1140,12 @@ public void DerivedPropertyAttribute()
...
+        public static void GetConverter_NetCoreApp(Type targetType, Type resultConverterType) =>
+            GetConverter(targetType, resultConverterType);
+
         class FooBarBase
```
Comment (jkotas) [inline src/libraries/System.ComponentModel.TypeConverter/tests/TypeDescriptorTests.cs]: You can also delete the `SkipOnTargetFramework` in this file while you are on it. It is no longer necessary.

File: src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj [hunk 3]
```diff
@@ -256,9 +256,9 @@
...
     <Compile Include="System\Text\Unicode\Utf8UtilityTests.ValidateBytes.cs" />
     <Compile Include="System\ArgIteratorTests.cs" />
-    <Compile Include="$(CommonPath)..\tests\System\RealFormatterTestsBase.netcoreapp.cs" Link="System\RealFormatterTestsBase.netcoreapp.cs" />
+    <Compile Include="$(CommonPath)..\tests\System\RealFormatterTestsBase.cs" Link="System\RealFormatterTestsBase.cs" />
     <Compile Include="System\RealFormatterTests.cs" />
-    <Compile Include="$(CommonPath)..\tests\System\RealParserTestsBase.netcoreapp.cs" Link="System\RealParserTestsBase.netcoreapp.cs" />
+    <Compile Include="$(CommonPath)..\tests\System\RealParserTestsBase.cs" Link="System\RealParserTestsBase.cs" />
     <Compile Include="System\RealParserTests.cs" />
   </ItemGroup>
   <ItemGroup Condition="'$(TargetsNetCoreApp)' != 'true'">
```
Comment (jkotas) [inline src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj]: You can delete this condition while you are on it.

**Comment by Marusyk** [comment]
@jkotas Thank you very much for all these reviews.
Consolidation is almost done ([except a few questions](https://github.com/dotnet/corefx/issues/41821#issuecomment-569140740))

**Review by jkotas — APPROVED** [review]
Thank you!

**Pull request merged**